src/js/pie.js

```diff
@@ -0,0 +1,182 @@
+                args: ["{that}"]
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "gpii.chartAuthoring.pieChart.pie.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    gpii.chartAuthoring.pieChart.pie.draw = function (that) {
+        var svg = that.svg,
+            pie = that.pie,
+            arc = that.arc,
+            color = that.color,
+            dataSet = that.model.dataSet,
+            sliceClass = that.classes.slice,
+            textClass = that.classes.text,
+            animationDuration = that.options.pieOptions.animationDuration;
+
+        var paths = svg.selectAll("path")
+            .data(pie(dataSet));
+        var texts = svg.selectAll("text")
+            .data(pie(dataSet));
+
+        // Draw pie slices
+        paths.enter()
+            .append("path")
+            .attr({
+                "fill": function(d, i) {
+                    return color(i);
+                },
+                "d": arc,
+                "class": sliceClass
+            })
+            .each(function (d) {
+                this._current = d;
+            });
+
+        paths.transition().duration(animationDuration).attrTween("d", function (a) {
+            var i = d3.interpolate(this._current, a);
+            this._current = i(0);
+            return function(t) {
+                return arc(i(t));
+            };
+        });
+
+        paths.exit().remove();
+
+        // Draw texts for pie slices
+        texts.enter()
+            .append("text")
+            .text(function(d) {
+                return d.value;
+            })
+            .attr({
+                "text-anchor": "middle",
+                "class": textClass,
+                "transform": function(d) {
+                    return "translate(" + arc.centroid(d) + ")";
+                }
+            });
+
+        // The same "transform" attribute definition for texts are used twice at the transition here and above when texts
+        // are created. This is because, the definition above pushes the text to the center of the pie slice when the
+        // text is created. The one at the transition here happs when a slice is tranformed from an old angle to a
+        // new angle and the duplicate transform moves the text from the center of the old slice to the center of the
+        // new slice. Having the first transform without the 2nd, the text would stay at the old position, the center
+        // of the old slice, without moving to the center of the new slice. Having the 2nd without the 1st, the transition
+        // would start from the center/origin of the pie circle instead of the center of the old slice.
+        texts.transition().duration(animationDuration).attr("transform", function(d) {
```

Comment:
I think, all the same, we can remove the duplication between the transforms. I suggest turning it into an invoker which accepts (d) and binds to {that}.arc in the definition